mne_bids/meg_bids.py

```diff
@@ -315,9 +317,17 @@ def raw_to_bids(subject_id, task, raw_file, output_path, session_id=None,
     data_meta_fname = make_bids_filename(
         subject=subject_id, session=session_id,
         suffix='%s.json' % kind, prefix=data_path)
-    raw_file_bids = make_bids_filename(
-        subject=subject_id, session=session_id, task=task, run=run,
-        suffix='%s%s' % (kind, ext), prefix=data_path)
+    if ext in ['.fif', '.gz', '.ds']:
+        raw_file_bids = make_bids_filename(
+            subject=subject_id, session=session_id, task=task, run=run,
+            suffix='%s%s' % (kind, ext))
+    else:
+        raw_folder = make_bids_filename(
+            subject=subject_id, session=session_id, task=task, run=run,
+            suffix='%s' % kind)
+        raw_file_bids = make_bids_filename(
```

Comment:
why do you need to do this? Shouldn't you be preserving the original filename if you have `raw_folder`?